service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -229,6 +234,18 @@ public DatasetPreviewTablesResponse listDatasetPreviewTables(
     return new DatasetPreviewTablesResponse().tables(tableMetadataList);
   }
 
+  private void validateSchema(String metaData) {
+    JsonNode node;
+    try {
+      node = objectMapper.readValue(metaData, JsonNode.class);
+    } catch (JsonProcessingException e) {
+      throw new IllegalMetadataException(e);
+    }
+    if (node.getNodeType() != JsonNodeType.OBJECT) {
```

Comment:
Instead of doing this check, you could instead call `readvalue()` with `ObjectNode.class` instead of `JsonNode.class`. Or, call `toJsonNode()` directly and add a user visible message to `toJsonNode()`.

Reply:
I think having the check makes it more clear what the intention of the code is, as opposed to one lining it

Reply:
If you used the name `validateStringIsJsonObject`, I would probably move to @pshapiro4broad's suggestion. Otherwise I agree with @tcjiang9, it is a little more clear this way without strong understanding of the internals of the library.

However, I might consider onelining it, and making the body of the try:
```
if (objectMapper.readValue(metadata, JsonNode.class).getNodeType() != JsonNodeType.OBJECT) {
  throw new BadRequestException("catalogEntry/metadata must be a json object");
}
```

Reply:
My thinking was that you should use the same API that's used by the code that retrieves the contents of the metadata and sends it back to the user. So if that code needs `ObjectNode`, this code should use it too. What if `getNodeType()` returns `JsonNodeType.OBJECT` but `readValue()` to `ObjectNode` fails?